parser/fileloader/jsonnetloader.go

```diff
@@ -21,17 +21,25 @@ type JsonnetLoader struct {
 }
 
 // LoadFile load file
-func (jl *JsonnetLoader) LoadFile(filePath string) (map[string]interface{}, error) {
+func (jl *JsonnetLoader) LoadFile(filePath string) ([]map[string]interface{}, error) {
 	inputFile, err := ioutil.ReadFile(filePath)
 	if err != nil {
 		return nil, err
 	}
-	evaluatedSnipped, err := jl.EvaluateSnippet(filePath, string(inputFile))
+	evaluatedSnippet, err := jl.EvaluateSnippetStream(filePath, string(inputFile))
 	if err != nil {
-		return nil, err
+    evaluatedSingleSnippet, err := jl.EvaluateSnippet(filePath, string(inputFile))
+    if err != nil {
+      return nil, err
+    }
+    evaluatedSnippet = []string{evaluatedSingleSnippet}
 	}
-	var output map[string]interface{}
-	json.Unmarshal([]byte(evaluatedSnipped), &output)
+	var output []map[string]interface{}
+  for i := range evaluatedSnippet {
```

Comment:
How’s this supposed to work? `evaluatedSnippet` contains an array of JSON strings, but after you successfully get `evaluatedSingleSnippet` you override this `evaluatedSnippet` array with `evaluatedSingleSnippet`. Therefore you called  `jl.EvaluateSnippetStream` in vain.

So I think you should remove all code within `if` statement above, except standard `return nil, err`

Reply:
also you should reformat code with gofmt

Reply:
EvaluateSnippetStream returns error if the jsonnet does not output an array of objects (this is the difference between `EvaluateSnippetStream` and `EvaluateSnippet`) and if it's not an array of objects, we treat it as an object and try to load it using `EvaluateSnippet`. If this fails as well, we most likely have an incorrect jsonnet file.

`EvaluateSnippetStream` is more strict in terms of output so we try this and if it fails, we go for generic "load anything" function and then `parseObjects` does the triaging.

Reply:
So if there is only one object in the file, then `EvaluateSnippetStream` fails? Hmm, it's strange behaviour but I understand now why you did it that way. So maybe small comment with explanation that that handles a case when there is only one object in the file would be a good idea?

Reply:
Added a comment on that, since it's not obvious. Hopefully we get to implement it better later :)